Arrowgene.Ddon.Database/Files/Database/Script/AddMissingOnDeleteCascades_migration.sql

```diff
@@ -1,4 +1,6 @@
-﻿PRAGMA foreign_keys=OFF;
+﻿COMMIT TRANSACTION;
```

Comment:
We definitely dont want to commit the transaction here if for some reason some later migration fails down the line and all the migrations have to be rolled back

Reply:
Shouldn't every migration be isolated? Why would we want to span the transaction across multiple migrations? Should we not just error out on the first failed migration and see every migration as a checkpoint?

Reply:
I would suggest something like this:

```
        public bool MigrateDatabase(IDatabase db, uint fromVersion, uint toVersion)
        {                
            List<IMigrationStrategy> strategies = FindStrategies(fromVersion, toVersion, out bool completesMigration);
            if(!completesMigration)
            {
                throw new Exception("No migration possible from "+fromVersion+" to "+toVersion);
            }

            foreach (var strategy in strategies)
            {
                bool successfullyMigrated = db.ExecuteInTransaction((conn) => {
                    strategy.Migrate(db, conn);
                });
                if (!successfullyMigrated)
                {
                    Logger.Exception(new Exception($"Failed to migrate database with strategy {strategy} from {fromVersion} to {toVersion}!"));
                    return false;
                }
            }
            return true;
        }
```

Reply:
Documented issue in discord, removed problematic intermediate commits from my migration here so we can proceed with merging.